scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -26,7 +26,7 @@ def _get_proxy(self, url, orig_type):
         proxy_url = urlunparse((proxy_type or orig_type, hostport, '', '', '', ''))
 
         if user:
-            user_pass = '%s:%s' % (unquote(user), unquote(password))
+            user_pass = to_bytes('%s:%s' % (unquote(user), unquote(password)))
```

Comment:
@kmike , indeed `Proxy-Authorization` value is defined in the same way as `Autorization`

https://www.rfc-editor.org/rfc/rfc2617.txt

```
3.6 Proxy-Authentication and Proxy-Authorization

   The digest authentication scheme may also be used for authenticating
   users to proxies, proxies to proxies, or proxies to origin servers by
   use of the Proxy-Authenticate and Proxy-Authorization headers. (...)
   The digest-challenge used in the
   Proxy-Authenticate header is the same as that for the WWW-
   Authenticate header as defined above in section 3.2.1.
```

https://www.rfc-editor.org/rfc/rfc7235.txt
```
Appendix C.  Collected ABNF
   Authorization = credentials
   Proxy-Authorization = credentials
```

so yes, it's saner to use a single code path.